server/game/cards/characters/07/grizzledminer.js

```diff
@@ -0,0 +1,26 @@
+const DrawCard = require('../../../drawcard.js');
+
+class GrizzledMiner extends DrawCard {
+
+    setupCardAbilities(ability) {
+        this.persistentEffect({
+            location: 'any',
+            targetType: 'player',
+            targetController: 'current',
+            effect: ability.effects.reduceSelfCost('marshal', this.getDiscount())
+        });
+    }
+
+    getDiscount() {
+        let cards = this.controller.filterCardsInPlay(
+            card => card.isFaction('thenightswatch')
+                && (card.getType() === 'attachment' || card.getType() === 'location'));
+
+        return Math.min(cards.length, this.getCost() - 1);
```

Comment:
OK, this should now (at last!) be done in the last update I've just pushed to this PR.